lib/src/collection/lru_map.dart

```diff
@@ -0,0 +1,286 @@
+part of quiver.collection;
+
+/**
+ * An implementation of a [Map] which has a maximum size and uses a Least
+ * Recently Used algorithm to remove items from the [Map] when the [maximumSize]
+ * is reached and new items are added.
+ */
+abstract class LruMap<K, V> implements Map<K, V> {
+  /**
+   * Creates a [LruMap] instance with the default implementation.
+   */
+  factory LruMap({int maximumSize}) = LinkedLruHashMap;
+
+  /**
+   * Maximum size of the [Map]. If [length] exceeds this value at any time, the
+   * oldest entries are removed beneath the current [length].
```

Comment:
s/the oldest entries/entries accessed the earliest/ maybe? "the oldest" makes it sound "added a long time ago".

It's probably my ESL, but the phrase "beneath the current length" sounds a bit fuzzy to me.